cluster-provision/gocli/cmd/run.go

```diff
@@ -869,48 +850,29 @@ func getDevicePCIID(pciAddress string) (string, error) {
 }
 
 // prepareDeviceForAssignment moves the deivce from it's original driver to vfio-pci driver
-func prepareDeviceForAssignment(cli *client.Client, nodeContainer, pciID, pciAddress string) error {
+func prepareDeviceForAssignment(sshClient libssh.Client, pciID, pciAddress string) error {
 	devicePCIID := pciID
 	if pciAddress != "" {
 		devicePCIID, _ = getDevicePCIID(pciAddress)
 	}
-	success, err := docker.Exec(cli, nodeContainer, []string{
-		"/bin/bash",
-		"-c",
-		fmt.Sprintf("ssh.sh sudo /bin/bash -s -- --vendor %s < /scripts/bind_device_to_vfio.sh", devicePCIID),
-	}, os.Stdout)
-	if err != nil {
+	if err := sshClient.Command(fmt.Sprintf("-s -- --vendor %s < /scripts/bind_device_to_vfio.sh", devicePCIID)); err != nil {
 		return err
 	}
-	if !success {
-		return fmt.Errorf("binding device to vfio driver failed")
-	}
 	return nil
 }
 
-func prepareEtcdDataMount(node string, etcdDataDir string, mountSize string) error {
-	var err error
-	var success bool
-
-	success, err = docker.Exec(cli, node, []string{"/bin/bash", "-c", fmt.Sprintf("ssh.sh sudo mkdir -p %s", etcdDataDir)}, os.Stdout)
-	if !success || err != nil {
-		return fmt.Errorf("create etcd data directory '%s'on node %s failed: %v", etcdDataDir, node, err)
-	}
-
-	success, err = docker.Exec(cli, node, []string{"/bin/bash", "-c", fmt.Sprintf("ssh.sh sudo test -d %s", etcdDataDir)}, os.Stdout)
-	if !success || err != nil {
-		return fmt.Errorf("verify etcd data directory '%s'on node %s exists failed: %v", etcdDataDir, node, err)
-	}
-
-	success, err = docker.Exec(cli, node, []string{"/bin/bash", "-c", fmt.Sprintf("ssh.sh sudo mount -t tmpfs -o size=%s tmpfs %s", mountSize, etcdDataDir)}, os.Stdout)
-	if !success || err != nil {
-		return fmt.Errorf("create tmpfs mount '%s' for etcd data on node %s failed: %v", etcdDataDir, node, err)
+func prepareEtcdDataMount(sshClient libssh.Client, etcdDataDir string, mountSize string) error {
+	cmds := []string{
+		fmt.Sprintf("mkdir -p %s", etcdDataDir),
+		fmt.Sprintf("test -d %s", etcdDataDir),
+		fmt.Sprintf("mount -t tmpfs -o size=%s tmpfs %s", mountSize, etcdDataDir),
```

Comment:
@aerosouund This needs to run with sudo, see failure in https://github.com/kubevirt/kubevirt/pull/12310